foreshadow/transformers/smart.py

```diff
@@ -52,17 +52,31 @@ class Encoder(SmartTransformer):
     """Automatically Encodes Categorical Features
 
     If there are less than 30 categories, then OneHotEncoder is used, if there are more
-    then HashingEncoder is used. If used in a y_var context, LabelEncoder is used.
+    then HashingEncoder is used. If the columns containing a delimmeter exceed delim_cuttoff then a
+    DummyEncoder is used (set cutoff to -1 to force). If used in a y_var context, LabelEncoder is used.
 
     """
 
-    def _get_transformer(self, X, y=None, unique_num_cutoff=30, **fit_params):
+    def _get_transformer(
+        self, X, y=None, unique_num_cutoff=30, delim_cutoff=0.5, **fit_params
```

Comment:
I am not sure using delim_cutoff will work. What could happen is majority of the values only have one category but the rest have delimited categories. It will also be hard to come up with a reasonable cutoff for this purpose.

An idea is to check after applying the delimiter, does it reduce the total number of categories. Here is an over-simplified example:

```python
a = pd.Series(['a,b', 'a', 'b'])
a
Out[10]:
0    a,b
1      a
2      b
```
Only 1/3 of samples have the delimiter, so it will fail the 0.5 cutoff. But after we apply the delimiter, it will only have 2 categories instead of 3. A criteria could be that resulted categories after applying delimiter should be less than 50% of the unique values of the column.

An idea of implementation is re-use the Series.str.get_dummies() to get the list of categories and compare it to the number of unique values in the column.